src/Elastic.Clients.Elasticsearch/_Shared/Api/SearchRequest.cs

```diff
@@ -36,6 +36,11 @@ protected override (string ResolvedUrl, string UrlTemplate, Dictionary<string, s
 [JsonConverter(typeof(SearchRequestOfTConverterFactory))]
 public partial class SearchRequest<TInferDocument> : SearchRequest
 {
+	static SearchRequest()
+	{
+		DynamicallyAccessed.PublicConstructors(typeof(SearchRequestOfTConverter<TInferDocument>));
```

Comment:
We might have to generate this for all generic types that are using converter factories.